package/MDAnalysis/topology/c_GROParser.pyx

```diff
@@ -0,0 +1,168 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding: utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""
+GRO topology parser
+===================
+
+Read a list of atoms from a GROMOS/Gromacs GRO coordinate file to
+build a basic topology.
+
+Atom types and masses are guessed.
+
+See Also
+--------
+:mod:`MDAnalysis.coordinates.GRO`
+
+
+
+Classes
+-------
+
+.. autoclass:: GROParser
+   :members:
+   :inherited-members:
+
+"""
+from __future__ import absolute_import
+
+import numpy as np
+cimport numpy as np
+from six.moves import range
+
+from ..lib.util import openany
+from ..core.topologyattrs import (
+    Atomnames,
+    Atomtypes,
+    Atomids,
+    Masses,
+    Resids,
+    Resnames,
+    Resnums,
+    Segids,
+)
+from ..core.topology import Topology
+from .base import TopologyReaderBase, change_squash
+from . import guessers
+
+
+class GROParser(TopologyReaderBase):
+    """Reads a Gromacs GRO file
+
+    Reads the following attributes:
+      - resids
+      - resnames
+      - atomids
+      - atomnames
+
+    Guesses the following attributes
+      - atomtypes
+      - masses
+    """
+    format = 'GRO'
+
+    def parse(self, **kwargs):
+        """Return the *Topology* object for this file"""
+        # Gro has the following columns
+        # resid, resname, name, index, (x,y,z)
+        cdef int n_atoms;
+        cdef int i;
+        cdef str line
+        cdef int count = 0
+        cdef np.ndarray[np.int32_t] resids
+        cdef np.ndarray[object] resnames
+        cdef np.ndarray[object] names
+        cdef np.ndarray[np.int32_t] indices
```

Comment:
better to use [typed memoryviews](https://cython.readthedocs.io/en/latest/src/userguide/memoryviews.html) here vs. the old buffer syntax